sdk/python/foundation-models/system/content-safety/score/score.py

```diff
@@ -0,0 +1,407 @@
+                    )
+                )
+except Exception as e:
+    _logger.warning(
+        "Failure processing model sample input: {}.\nWill attempt to create sample input based on model signature. "
+        "For more information, please see: https://aka.ms/aml-mlflow-deploy.".format(e)
+    )
+
+# Handle the signature information to attempt creation of a sample based on signature if no concrete
+# sample was provided
+model_signature = model.signature
+if model_signature:
+    model_signature_inputs = model_signature.inputs
+    model_signature_outputs = model_signature.outputs
+    if model_signature_inputs and sample_input is None:
+        if model_signature_inputs.is_tensor_spec():
+            sample_input = create_tensor_spec_sample_io(model_signature_inputs)
+        else:
+            sample_input = create_col_spec_sample_io(model_signature_inputs)
+
+    if model_signature_outputs and sample_output is None:
+        if model_signature_outputs.is_tensor_spec():
+            sample_output = create_tensor_spec_sample_io(model_signature_outputs)
+        else:
+            sample_output = create_col_spec_sample_io(model_signature_outputs)
+else:
+    _logger.warning(
+        "No signature information provided for model. If no sample information was provided with the model "
+        "the deployment's swagger will not include input and output schema and typing information."
+        "For more information, please see: https://aka.ms/aml-mlflow-deploy."
+    )
+
+if sample_input is None:
+    input_param = NoSampleParameterType()
+else:
+    if isinstance(sample_input, np.ndarray):
+        # Unnamed tensor input
+        input_param = NumpyParameterType(sample_input, enforce_shape=False)
+    elif isinstance(sample_input, dict):
+        param_arg = {}
+        for key, value in sample_input.items():
+            param_arg[key] = NumpyParameterType(value, enforce_shape=False)
+        input_param = StandardPythonParameterType(param_arg)
+    else:
+        input_param = PandasParameterType(
+            sample_input, enforce_shape=False, orient="split"
+        )
+
+if sample_output is None:
+    output_param = NoSampleParameterType()
+else:
+    if isinstance(sample_output, np.ndarray):
+        # Unnamed tensor input
+        output_param = NumpyParameterType(sample_output, enforce_shape=False)
+    elif isinstance(sample_output, dict):
+        param_arg = {}
+        for key, value in sample_output.items():
+            param_arg[key] = NumpyParameterType(value, enforce_shape=False)
+        output_param = StandardPythonParameterType(param_arg)
+    else:
+        output_param = PandasParameterType(
+            sample_output, enforce_shape=False, orient="records"
+        )
+
+# we use mlflow.pyfunc's load_model function because it has a predict function on it we need for inferencing
+model = load_model(model_path)
+
+
+def init():
+    global aacs_client
+    endpoint = os.environ.get("CONTENT_SAFETY_ENDPOINT")
```

Comment:
Content safety support should eventually be in the default NCD scoring script, what is our plan to merge this into the regular NCD flow?